packages/driver/src/cy/commands/actions/type.coffee

```diff
@@ -112,6 +112,9 @@ module.exports = (Commands, Cypress, cy, state, config) ->
       if chars is ""
         $utils.throwErrByPath("type.empty_string", { onFail: options._log })
 
+      if options.$el.attr("maxlength") and chars.length > options.$el.attr("maxlength")
```

Comment:
`options.$el.attr("maxlength")` returns a string. We should `parseInt` the value and ignore if the result is not a number. That would consistent with Chrome's behavior, at least. `maxlength="5foo"` results in a limit of 5 characters. `maxlength="foo"` just gets ignored.